File: Example/STCardsMenu-ObjC/STCardsMenu-ObjC/ViewController.m
//
//  ViewController.m
//  STCardsMenu-ObjC
//
//  Created by TangJR on 12/14/15.
//  Copyright © 2015 tangjr. All rights reserved.
//

#import "ViewController.h"
#import "STCardsMenuRootVC.h"
#import "UIColor+STCreation.h"
#import "STCardsMenuChildVC.h"
#import "RootViewController.h"
#import "ChildViewController.h"

@interface ViewController ()

@property (weak, nonatomic) STCardsMenuRootVC *cardsMenuRootVC;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    RootViewController *rootVC = [RootViewController new];
    [self.view addSubview:rootVC.view];
    [self addChildViewController:rootVC];
    self.cardsMenuRootVC = rootVC;
    
    [self setupChildViewControllers];
}

- (void)setupChildViewControllers {
    NSArray *backgroundColors = @[[UIColor st_colorWithHex:0x553860], [UIColor st_colorWithHex:0xA5435D], [UIColor st_colorWithHex:0xF18264], [UIColor st_colorWithHex:0xFEE177]];
    NSArray *titles = @[@"CONTACT", @"ABOUT", @"TEAM", @"PROJECTS"];
    NSArray *tintColors = @[[UIColor st_colorWithHex:0xF18264], [UIColor st_colorWithHex:0xFEE177], [UIColor st_colorWithHex:0x553860], [UIColor st_colorWithHex:0xA5435D]];
    for (int i = 0; i < 4; i++) {
        ChildViewController *childVC = [ChildViewController new];
        childVC.tintColor = tintColors[i];
        childVC.title = titles[i];
        childVC.view.backgroundColor = backgroundColors[i];
        [self.cardsMenuRootVC.view addSubview:childVC.view];
        [self.cardsMenuRootVC addChildViewController:childVC];
    }
}

@end

File: Example/STCardsMenu-Swift/STCardsMenu-Swift/ViewController.swift
//
//  ViewController.swift
//  STCardsMenu-Swift
//
//  Created by TangJR on 12/15/15.
//  Copyright © 2015 tangjr. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    let rootVC = RootViewController()

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        addChildViewController(rootVC)
        view.addSubview(rootVC.view)
        
        setupChildViewControllers()
    }
    
    private func setupChildViewControllers() {
        let titles = ["CONTACT", "ABOUT", "TEAM", "PROJECTS"]
        let tintColors = [UIColor(st_hex: 0xF18264), UIColor(st_hex: 0xFEE177), UIColor(st_hex: 0x553860), UIColor(st_hex: 0xA5435D)]
        let backgroundColors = [UIColor(st_hex: 0x553860), UIColor(st_hex: 0xA5435D), UIColor(st_hex: 0xF18264), UIColor(st_hex: 0xFEE177)]
        for i in 0 ..< 4 {
            let childVC = ChildViewController()
            childVC.view.backgroundColor = backgroundColors[i]
            childVC.title = titles[i]
            childVC.tintColor = tintColors[i]
            rootVC.view.addSubview(childVC.view)
            rootVC.addChildViewController(childVC)
        }
    }
}